Sources/SwiftDocC/Infrastructure/Link Resolution/PathHierarchyBasedLinkResolver+Breadcrumbs.swift

```diff
@@ -0,0 +1,56 @@
+/*
+ This source file is part of the Swift.org open source project
+
+ Copyright (c) 2024 Apple Inc. and the Swift project authors
+ Licensed under Apache License v2.0 with Runtime Library Exception
+
+ See https://swift.org/LICENSE.txt for license information
+ See https://swift.org/CONTRIBUTORS.txt for Swift project authors
+*/
+
+import Foundation
+import SymbolKit
+
+extension PathHierarchyBasedLinkResolver {
+    
+    /// Finds the canonical path, also called "breadcrumb", to the given symbol in the path hierarchy.
+    /// 
+    /// The path is a list of references that describe a walk through the path hierarchy descending from the module down to, but not including, the given `reference`.
+    ///
+    /// - Parameters:
+    ///   - reference: The symbol reference to find the canonical path to.
+    ///   - sourceLanguage: The source language representation of the symbol to fin the canonical path for.
+    /// - Returns: The canonical path to the given symbol reference, or `nil` if the reference isn't a symbol or if the symbol doesn't have a representation in the given source language.
+    func breadcrumbs(of reference: ResolvedTopicReference, in sourceLanguage: SourceLanguage) -> [ResolvedTopicReference]? {
```

Comment:
Maybe use a singular name for this since it only returns a single breadcrumb, not a list of breadcrumbs.

Reply:
I think I consider a path to be a list of breadcrumbs and a single breadcrumb to only be one path component.

I see that I used singular "breadcrumb" in the documentation comment. I'll update that documentation to say "breadcrumbs"

Reply:
I added a code comment in https://github.com/swiftlang/swift-docc/pull/1081/commits/8647adf5976dc37a7304eb537d6fc0e57af9bc8a